network/crl.go

```diff
@@ -0,0 +1,389 @@
+	if err != nil {
+		log.WithError(err).Debugf("CRL: Cannot parse CRL from '%s'", url)
+		return nil, err
+	}
+
+	err = issuerCert.CheckCRLSignature(crl)
+	if err != nil {
+		log.WithError(err).Warnf("CRL: Failed to check signature for CRL at %s", url)
+		return nil, err
+	}
+
+	if crl.TBSCertList.NextUpdate.Before(time.Now()) {
+		log.Warnf("CRL: CRL at %s is outdated", url)
+		return nil, ErrOutdatedCRL
+	}
+
+	if v.Config.isCachingEnabled() {
+		cacheItem := &CRLCacheItem{Fetched: time.Now(), CRL: *crl}
+		v.Cache.Put(url, cacheItem)
+	}
+
+	return crl, nil
+}
+
+func (v DefaultCRLVerifier) verifyCertWithIssuer(cert, issuer *x509.Certificate) error {
+	log.Debugf("CRL: Verifying '%s'", cert.Subject.String())
+
+	if len(v.Config.url) == 0 && v.Config.fromCert == crlFromCertIgnore {
+		log.Debugln("CRL: Skipping check since no config URL specified and we were told to ignore URLs from certificate")
+		return nil
+	}
+
+	for _, crlDistributionPoint := range cert.CRLDistributionPoints {
+		log.Debugf("CRL: certificate contains CRL URL: %s", crlDistributionPoint)
+	}
+
+	queriedCRLs := make(map[string]struct{})
+
+	if v.Config.url != "" {
+		crl, err := v.getCachedOrFetch(v.Config.url, issuer)
+		if err != nil {
+			log.WithError(err).Debugf("CRL: Cannot get CRL from '%s'", v.Config.url)
+			return err
+		}
+
+		for _, revokedCertificate := range crl.TBSCertList.RevokedCertificates {
+			log.WithField("extensions", revokedCertificate.Extensions).Warnln("Revoked certificate extensions")
+			if revokedCertificate.SerialNumber.Cmp(cert.SerialNumber) == 0 {
+				log.Warnf("CRL: Certificate %v was revoked at %v", cert.SerialNumber, revokedCertificate.RevocationTime)
+				return ErrCertWasRevoked
+			}
+		}
+
+		queriedCRLs[v.Config.url] = struct{}{}
+		log.Debugln("CRL: OK, not found in list of revoked certificates")
+	}
+
+	if len(cert.CRLDistributionPoints) > 0 && v.Config.fromCert != crlFromCertIgnore {
+		for _, crlDistributionPoint := range cert.CRLDistributionPoints {
+			if _, ok := queriedCRLs[crlDistributionPoint]; ok {
+				continue
+			}
+
+			log.Debugf("CRL: using '%s' from certificate", crlDistributionPoint)
+
+			crl, err := v.getCachedOrFetch(crlDistributionPoint, issuer)
+			if err != nil {
+				log.WithError(err).Debugf("CRL: Cannot get CRL from '%s'", crlDistributionPoint)
+				return err
+			}
+
+			for _, revokedCertificate := range crl.TBSCertList.RevokedCertificates {
+				if revokedCertificate.SerialNumber.Cmp(cert.SerialNumber) == 0 {
+					log.Warnf("CRL: Certificate %v was revoked at %v", cert.SerialNumber, revokedCertificate.RevocationTime)
+					return ErrCertWasRevoked
+				}
+			}
+
+			queriedCRLs[v.Config.url] = struct{}{}
+			log.Debugln("CRL: OK, not found in list of revoked certificates")
+		}
+	}
+
+	return nil
+}
+
+// Verify ensures configured CRLs do not contain certificate from passed chain
+func (v DefaultCRLVerifier) Verify(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error {
```

Comment:
I do not see `rawCerts` using in this function. Do you need it at al?

Reply:
As explained [here](https://github.com/cossacklabs/acra/pull/436#discussion_r539399797), we don't need it, left it for compatibility and future extensions